cmd/src/debug_common.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sourcegraph/src-cli/internal/exec"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type archiveFile struct {
+	name string
+	data []byte
+	err  error
+}
+
+func archiveFileFromCommand(ctx context.Context, path, cmd string, args ...string) *archiveFile {
+	f := &archiveFile{name: path}
+	f.data, f.err = exec.CommandContext(ctx, cmd, args...).CombinedOutput()
+	if f.err != nil {
+		f.err = errors.Wrapf(f.err, "executing command: %s %s: received error: %s", cmd, strings.Join(args, " "), f.data)
+	}
+	return f
+}
+
+// This function prompts the user to confirm they want to run the command
+func verify(confirmationText string) (bool, error) {
+	input := ""
+	for strings.ToLower(input) != "y" && strings.ToLower(input) != "n" {
+		fmt.Printf("%s [y/N]: ", confirmationText)
+		if _, err := fmt.Scanln(&input); err != nil {
+			return false, err
+		}
+	}
+
+	return strings.ToLower(input) == "y", nil
+}
+
+func processBaseDir(base string) (string, string) {
+	var baseDir string
+	if !strings.HasSuffix(base, ".zip") {
+		baseDir = base
+		base = base + ".zip"
+	} else {
+		baseDir = strings.TrimSuffix(base, ".zip")
+	}
+	return base, baseDir
```

Comment:
```suggestion
	if !strings.HasSuffix(base, ".zip") {
		return base + ".zip", base
	}

	return base, strings.TrimSuffix(base, ".zip")
```

For small one-or-the-other functions it's usually nicer to just handle one case conditionally, then the other like this. Go will actually issue a lint if you do an `else` where the previous if consequent returns.